grafanalib/zabbix.py

```diff
@@ -0,0 +1,706 @@
+import re
+import attr
+from attr.validators import instance_of
+
+
+ZABBIX_QMODE_METRICS = 0
+ZABBIX_QMODE_SERVICES = 1
+ZABBIX_QMODE_TEXT = 2
+
+ZABBIX_SLA_PROP_STATUS = {
+    "name": "Status",
+    "property": "status"}
+
+ZABBIX_SLA_PROP_SLA = {
+    "name": "SLA",
+    "property": "sla"}
+
+ZABBIX_SLA_PROP_OKTIME = {
+    "name": "OK time",
+    "property": "okTime"}
+
+ZABBIX_SLA_PROP_PROBTIME = {
+    "name": "Problem time",
+    "property": "problemTime"}
+
+ZABBIX_SLA_PROP_DOWNTIME = {
+    "name": "Down time",
+    "property": "downtimeTime"}
+
+
+@attr.attributes(repr=False, slots=True)
+class _IsInValidator(object):
+    choices = attr.attr()
+
+    def __call__(self, inst, attr, value):
+        if value not in self.choices:
+            raise ValueError("{attr} should be one of {choice}".format(
+                attr=attr, choice=self.choices))
+
+    def __repr__(self):
+        return (
+            "<is value present in list of  {choice}>"
+            .format(choice=self.choices)
+        )
+
+
+def is_in_validator(choices):
+    """
+    A validator that raises a :exc:`ValueError` if the attribute value is not
+    in a provided list.
+
+    :param choices: List of valid choices
+    """
+    return _IsInValidator(choices)
+
+
+def is_valid_interval(instance, attribute, value):
+    """
+    A validator that raises a :exc:`ValueError` if the attribute value is not
+    matching on regular expression.
+    """
+    if not re.match("^[+-]?\d*[smhdMY]$", value):
+        raise ValueError(
+            "valid interval should be a string "
+            "matching an expressiong: ^[+-]?\d*[smhdMY]$. "
+            "Exaxples: 24h 7d 1M +24h -24h")
+
+
+def is_number(instance, attribute, value):
+    """
+    A validator that raises a :exc:`ValueError` if the attribute value is not
+    a number (not int or float).
+    """
+    if isinstance(value, int) or isinstance(value, float):
```

Comment:
This function probably isn't necessary. Can just do `instance_of(Number)`, see https://docs.python.org/2/library/numbers.html#numbers.Number for details.